synapse/federation/federation_server.py

```diff
@@ -450,6 +451,7 @@ async def _on_state_ids_request_compute(self, room_id, event_id):
     async def _on_context_state_request_compute(
         self, room_id: str, event_id: str
     ) -> Dict[str, list]:
+        pdus = []  # type: Iterable[EventBase]
```

Comment:
this is needed, because else mypy will complain about the `pdu =` in the second branch not being `List[EventBase]`